ui/codeView.js

```diff
@@ -89,8 +92,67 @@ function _getWindowId(win) {
     return 'window:%d'.format(win.get_stable_sequence());
 }
 
+function _getHackToolboxProxy(win) {
+    let hackToolboxId = HACK_TOOLBOX_ID;
+
+    // This will work only on EndlessOS
+    if (Shell.WindowTracker.get_hack_toolbox_proxy) {
+        return Shell.WindowTracker.get_hack_toolbox_proxy(win);
+    }
```

Comment:
Is it possible to not rely on `get_hack_toolbox_proxy` anymore? If we can drop the patch that adds the hack toolbox, it'll be better.

Reply:
Yes, I think so. This patch provides a fallback behavior for this so I think we drop that patch. There are some issues yet, with windows pairing (app + toolbox), but I'll fix that overriding some functions on the dash icon and overview mode.